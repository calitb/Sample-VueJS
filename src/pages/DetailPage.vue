<template>
  <div v-if="item">
    <b-card
      :title="item.name"
      :img-src="item | imageSRC"
      img-alt="Image"
      img-top
      style="max-width: 20rem;"
      class="mb-2"
    ></b-card>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { Item } from '@/items';

export default Vue.extend({
  name: 'app-detalle',
  props: {
    detailId: String
  },
  computed: {
    item(): Item | undefined {
      return this.$root.$data.items.find((item: Item) => parseInt(item.id, 10) === parseInt(this.detailId, 10));
    }
  },
  created() {
    if (isNaN(Number(this.detailId))) {
      this.$router.push({ name: 'error' });
    }
  }
});
</script>
