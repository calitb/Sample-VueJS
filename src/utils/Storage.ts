export function removeItem(key: string) {
  sessionStorage.removeItem(key);
}

export function getItem(key: string): string | null {
  return sessionStorage.getItem(key);
}

export function setItem(key: string, value: string) {
  sessionStorage.setItem(key, value);
}
