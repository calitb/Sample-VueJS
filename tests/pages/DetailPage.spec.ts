import { shallowMount } from "@vue/test-utils";
import DetailPage from "@/pages/DetailPage.vue";
import Vuex from "vuex";

import { createLocalVue } from "../utils";
const localVue = createLocalVue();

describe("Detail Page", () => {
  describe("Default", () => {
    const store = new Vuex.Store({
      getters: {
        currentItem() {
          return { id: "4", name: "Charmander" };
        }
      }
    });

    const wrapper = shallowMount(DetailPage, {
      store,
      localVue
    });

    it("should render the component", () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it("should render the image", () => {
      expect(
        wrapper
          .find("article")
          .find("img")
          .attributes().src
      ).toBe("https://img.pokemondb.net/artwork/charmander.jpg");
    });

    it("should render the name", () => {
      expect(
        wrapper
          .find("article")
          .find("p")
          .text()
      ).toBe("Charmander");
    });
  });

  describe("Wrong route", () => {
    const localVue = createLocalVue();
    const store = new Vuex.Store({
      getters: {
        currentItem() {
          return undefined;
        }
      }
    });

    const wrapper = shallowMount(DetailPage, {
      store,
      localVue
    });

    it("should render the component", () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it("should render an error message", () => {
      const content = wrapper.find("div");
      expect(content.text()).toBe("NO EXISTE");
    });
  });
});
