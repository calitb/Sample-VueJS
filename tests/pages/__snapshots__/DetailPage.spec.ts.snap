// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Page Default should render the component 1`] = `
<div>
  <b-card-stub
    bodytag="div"
    class="mb-2"
    footertag="div"
    headertag="div"
    imgalt="Image"
    imgsrc="https://img.pokemondb.net/artwork/charmander.jpg"
    imgtop="true"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="div"
    title="Charmander"
    titletag="h4"
  />
</div>
`;

exports[`Detail Page Default should render the component 2`] = `
<div>
  <b-card-stub
    bodytag="div"
    class="mb-2"
    footertag="div"
    headertag="div"
    imgalt="Image"
    imgsrc="https://img.pokemondb.net/artwork/charmander.jpg"
    imgtop="true"
    subtitletag="h6"
    subtitletextvariant="muted"
    tag="div"
    title="Charmander"
    titletag="h4"
  />
</div>
`;
